distsql/distsql.go

```diff
@@ -141,46 +144,57 @@ func (r *selectResult) NextRaw() ([]byte, error) {
 }
 
 // NextChunk reads data to the chunk.
-func (r *selectResult) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) error {
+func (r *selectResult) NextChunk(goCtx goctx.Context, chk *chunk.Chunk) (int64, error) {
 	chk.Reset()
+	count := int64(0)
 	for chk.NumRows() < r.ctx.GetSessionVars().MaxChunkSize {
 		if r.selectResp == nil || r.respChkIdx == len(r.selectResp.Chunks) {
-			err := r.getSelectResp()
-			if err != nil || r.selectResp == nil {
-				return errors.Trace(err)
+			cnt, err := r.getSelectResp()
```

Comment:
no need to change the signature of  function `getSelectResp()`, we can just `return chk.NumRows(), nil`.

Reply:
What we need is the count of row scanned, not the returned.